CompanyApi/Controllers/CompanyController.cs

```diff
@@ -22,8 +22,85 @@ public ActionResult<Company> Create(CreateCompanyRequest request)
 
         [HttpDelete]
         public void ClearData()
-        { 
+        {
             companies.Clear();
         }
+
+        [HttpGet]
+        public ActionResult<List<Company>> Get()
+        {
+            return Ok(companies);
+        }
+
+        [HttpGet("{id}")]
+        public ActionResult<Company> GetById(string id)
+        {
+            Company? company = companies.FirstOrDefault(c => c.Id.Equals(id));
+            if (company == null)
+            {
+                return NotFound();
+            }
+            return Ok(company);
+        }
+
+        [HttpGet("details")]
+        public ActionResult<List<Company>> GetXCompaniesFromYPage(int pageSize, int pageIndex)
+        {
+            int startIndex = (pageIndex-1) * pageSize;
+            int endIndex = pageIndex * pageSize > companies.Count ? companies.Count : pageIndex * pageSize;
+            if (startIndex >= companies.Count)

```

Comment:
It is best not to modify the path provided by the documentation, you can use FromQuery to pass in parameters.